The `reflect-config.json` file isn't generated if the `io.micronaut.application` plugin is applied

### Expected Behavior

A Micronaut Data JPA application with GraalVM should work.

### Actual Behaviour

Micronaut Data JPA with GraalVM test applications have started to fail this morning. They worked last Friday.

When starting the application it fails with:

```
 __  __ _                                  _   
|  \/  (_) ___ _ __ ___  _ __   __ _ _   _| |_ 
| |\/| | |/ __| '__/ _ \| '_ \ / _` | | | | __|
| |  | | | (__| | | (_) | | | | (_| | |_| | |_ 
|_|  |_|_|\___|_|  \___/|_| |_|\__,_|\__,_|\__|
  Micronaut (v3.2.1-SNAPSHOT)

12:57:11.892 [main] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Starting...
12:57:11.899 [main] INFO  com.zaxxer.hikari.HikariDataSource - HikariPool-1 - Start completed.
12:57:11.899 [main] INFO  org.hibernate.Version - HHH000412: Hibernate ORM core version 5.5.3.Final
12:57:11.901 [main] INFO  o.h.annotations.common.Version - HCANN000001: Hibernate Commons Annotations {5.1.2.Final}
12:57:11.902 [main] INFO  org.hibernate.dialect.Dialect - HHH000400: Using dialect: org.hibernate.dialect.PostgreSQL95Dialect
12:57:11.903 [main] ERROR i.m.c.h.jpa.EntityManagerFactoryBean - Hibernate mapping error
org.hibernate.AnnotationException: No identifier specified for entity: example.domain.Pet
	at org.hibernate.cfg.InheritanceState.determineDefaultAccessType(InheritanceState.java:266)
	at org.hibernate.cfg.InheritanceState.getElementsToProcess(InheritanceState.java:211)
	at org.hibernate.cfg.AnnotationBinder.bindClass(AnnotationBinder.java:783)
	at org.hibernate.boot.model.source.internal.annotations.AnnotationMetadataSourceProcessorImpl.processEntityHierarchies(AnnotationMetadataSourceProcessorImpl.java:225)
	at org.hibernate.boot.model.process.spi.MetadataBuildingProcess$1.processEntityHierarchies(MetadataBuildingProcess.java:239)
	at org.hibernate.boot.model.process.spi.MetadataBuildingProcess.complete(MetadataBuildingProcess.java:282)
	at org.hibernate.boot.model.process.spi.MetadataBuildingProcess.build(MetadataBuildingProcess.java:86)
	at org.hibernate.boot.internal.MetadataBuilderImpl.build(MetadataBuilderImpl.java:479)
	at org.hibernate.boot.internal.MetadataBuilderImpl.build(MetadataBuilderImpl.java:85)
	at org.hibernate.boot.MetadataSources.buildMetadata(MetadataSources.java:202)
	at io.micronaut.configuration.hibernate.jpa.EntityManagerFactoryBean.hibernateSessionFactoryBuilder(EntityManagerFactoryBean.java:188)
	at io.micronaut.configuration.hibernate.jpa.$EntityManagerFactoryBean$HibernateSessionFactoryBuilder2$Definition.build(Unknown Source)
	at io.micronaut.context.BeanDefinitionDelegate.build(BeanDefinitionDelegate.java:136)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2337)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingletonInternal(DefaultBeanContext.java:3282)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingleton(DefaultBeanContext.java:3268)
	at io.micronaut.context.DefaultBeanContext.getBeanForDefinition(DefaultBeanContext.java:2821)
	at io.micronaut.context.DefaultBeanContext.getBeanInternal(DefaultBeanContext.java:2783)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1639)
	at io.micronaut.context.AbstractInitializableBeanDefinition.resolveBean(AbstractInitializableBeanDefinition.java:1933)
	at io.micronaut.context.AbstractInitializableBeanDefinition.getBeanForConstructorArgument(AbstractInitializableBeanDefinition.java:1189)
	at io.micronaut.configuration.hibernate.jpa.$EntityManagerFactoryBean$HibernateSessionFactory3$Definition.build(Unknown Source)
	at io.micronaut.context.BeanDefinitionDelegate.build(BeanDefinitionDelegate.java:136)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2337)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingletonInternal(DefaultBeanContext.java:3282)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingleton(DefaultBeanContext.java:3268)
	at io.micronaut.context.DefaultBeanContext.loadContextScopeBean(DefaultBeanContext.java:2654)
	at io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1933)
	at io.micronaut.context.DefaultApplicationContext.initializeContext(DefaultApplicationContext.java:237)
	at io.micronaut.context.DefaultBeanContext.readAllBeanDefinitionClasses(DefaultBeanContext.java:3454)
	at io.micronaut.context.DefaultBeanContext.start(DefaultBeanContext.java:330)
	at io.micronaut.context.DefaultApplicationContext.start(DefaultApplicationContext.java:183)
	at io.micronaut.runtime.Micronaut.start(Micronaut.java:72)
	at io.micronaut.runtime.Micronaut.run(Micronaut.java:313)
	at io.micronaut.runtime.Micronaut.run(Micronaut.java:299)
	at example.Application.main(Application.java:30)
12:57:11.903 [main] ERROR io.micronaut.runtime.Micronaut - Error starting Micronaut server: Bean definition [org.hibernate.SessionFactory] could not be loaded: Error instantiating bean of type  [org.hibernate.SessionFactory]

Message: No identifier specified for entity: example.domain.Pet
Path Taken: SessionFactory.hibernateSessionFactory([SessionFactoryBuilder sessionFactoryBuilder])
io.micronaut.context.exceptions.BeanInstantiationException: Bean definition [org.hibernate.SessionFactory] could not be loaded: Error instantiating bean of type  [org.hibernate.SessionFactory]

Message: No identifier specified for entity: example.domain.Pet
Path Taken: SessionFactory.hibernateSessionFactory([SessionFactoryBuilder sessionFactoryBuilder])
	at io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1939)
	at io.micronaut.context.DefaultApplicationContext.initializeContext(DefaultApplicationContext.java:237)
	at io.micronaut.context.DefaultBeanContext.readAllBeanDefinitionClasses(DefaultBeanContext.java:3454)
	at io.micronaut.context.DefaultBeanContext.start(DefaultBeanContext.java:330)
	at io.micronaut.context.DefaultApplicationContext.start(DefaultApplicationContext.java:183)
	at io.micronaut.runtime.Micronaut.start(Micronaut.java:72)
	at io.micronaut.runtime.Micronaut.run(Micronaut.java:313)
	at io.micronaut.runtime.Micronaut.run(Micronaut.java:299)
	at example.Application.main(Application.java:30)
Caused by: io.micronaut.context.exceptions.BeanInstantiationException: Error instantiating bean of type  [org.hibernate.SessionFactory]

Message: No identifier specified for entity: example.domain.Pet
Path Taken: SessionFactory.hibernateSessionFactory([SessionFactoryBuilder sessionFactoryBuilder])
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2364)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingletonInternal(DefaultBeanContext.java:3282)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingleton(DefaultBeanContext.java:3268)
	at io.micronaut.context.DefaultBeanContext.getBeanForDefinition(DefaultBeanContext.java:2821)
	at io.micronaut.context.DefaultBeanContext.getBeanInternal(DefaultBeanContext.java:2783)
	at io.micronaut.context.DefaultBeanContext.getBean(DefaultBeanContext.java:1639)
	at io.micronaut.context.AbstractInitializableBeanDefinition.resolveBean(AbstractInitializableBeanDefinition.java:1933)
	at io.micronaut.context.AbstractInitializableBeanDefinition.getBeanForConstructorArgument(AbstractInitializableBeanDefinition.java:1189)
	at io.micronaut.configuration.hibernate.jpa.$EntityManagerFactoryBean$HibernateSessionFactory3$Definition.build(Unknown Source)
	at io.micronaut.context.BeanDefinitionDelegate.build(BeanDefinitionDelegate.java:136)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2337)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingletonInternal(DefaultBeanContext.java:3282)
	at io.micronaut.context.DefaultBeanContext.createAndRegisterSingleton(DefaultBeanContext.java:3268)
	at io.micronaut.context.DefaultBeanContext.loadContextScopeBean(DefaultBeanContext.java:2654)
	at io.micronaut.context.DefaultBeanContext.initializeContext(DefaultBeanContext.java:1933)
	... 8 common frames omitted
Caused by: org.hibernate.AnnotationException: No identifier specified for entity: example.domain.Pet
	at org.hibernate.cfg.InheritanceState.determineDefaultAccessType(InheritanceState.java:266)
	at org.hibernate.cfg.InheritanceState.getElementsToProcess(InheritanceState.java:211)
	at org.hibernate.cfg.AnnotationBinder.bindClass(AnnotationBinder.java:783)
	at org.hibernate.boot.model.source.internal.annotations.AnnotationMetadataSourceProcessorImpl.processEntityHierarchies(AnnotationMetadataSourceProcessorImpl.java:225)
	at org.hibernate.boot.model.process.spi.MetadataBuildingProcess$1.processEntityHierarchies(MetadataBuildingProcess.java:239)
	at org.hibernate.boot.model.process.spi.MetadataBuildingProcess.complete(MetadataBuildingProcess.java:282)
	at org.hibernate.boot.model.process.spi.MetadataBuildingProcess.build(MetadataBuildingProcess.java:86)
	at org.hibernate.boot.internal.MetadataBuilderImpl.build(MetadataBuilderImpl.java:479)
	at org.hibernate.boot.internal.MetadataBuilderImpl.build(MetadataBuilderImpl.java:85)
	at org.hibernate.boot.MetadataSources.buildMetadata(MetadataSources.java:202)
	at io.micronaut.configuration.hibernate.jpa.EntityManagerFactoryBean.hibernateSessionFactoryBuilder(EntityManagerFactoryBean.java:188)
	at io.micronaut.configuration.hibernate.jpa.$EntityManagerFactoryBean$HibernateSessionFactoryBuilder2$Definition.build(Unknown Source)
	at io.micronaut.context.BeanDefinitionDelegate.build(BeanDefinitionDelegate.java:136)
	at io.micronaut.context.DefaultBeanContext.doCreateBean(DefaultBeanContext.java:2337)
	... 22 common frames omitted
```

This happens with all databases except H2.

### Steps To Reproduce

- `git clone https://github.com/micronaut-graal-tests/micronaut-data-jpa-graal`
- `cd micronaut-data-jpa-graal`
- `git checkout error-graalvm`
- `sdk use java 21.3.0.r11-grl`
- `./gradlew nativeCompile`
- `docker run -it --rm -p 5432:5432 -e POSTGRES_PASSWORD=secret -e POSTGRES_DB=pets postgres:11.5-alpine`
- `./build/native/nativeCompile/data-jpa-postgres`


### Environment Information

- Linux Mint 20.2
- GraalVM JDK 11 & 17

### Example Application

_No response_

### Version

3.2.1-SNAPSHOT and Micronaut Data 3.2.1-SNAPSHOT

--- a/build.gradle
+++ b/build.gradle
@@ -7,6 +7,7 @@ plugins {
     id "groovy"
     id "com.adarshr.test-logger"
     id "io.micronaut.internal.build.documented"
+    id "org.gradle.test-retry" version "1.3.1"
 }
 
 // If your plugin has any external java dependencies, Gradle will attempt to
@@ -62,7 +63,12 @@ dependencies {
     testImplementation libs.spock.junit4
 }
 
-test {
+tasks.withType(Test).configureEach {
+    retry {
+        maxRetries = 2
+        maxFailures = 20
+        failOnPassedAfterRetry = true
+    }
     useJUnitPlatform()
 }
 // The configuration example below shows the minimum required properties

--- a/gradle/libs.versions.toml
+++ b/gradle/libs.versions.toml
@@ -5,7 +5,7 @@ diffplug = "3.33.2"
 shadow = "7.1.0"
 groovy = "3.0.9"
 spock = "2.0-groovy-3.0"
-graalvmPlug = "0.9.7.1"
+graalvmPlug = "0.9.8"
 
 [libraries]
 dockerPlug = { module = "com.bmuschko:gradle-docker-plugin", version.ref = "docker" }

--- a/src/main/java/io/micronaut/gradle/docker/NativeImageDockerfile.java
+++ b/src/main/java/io/micronaut/gradle/docker/NativeImageDockerfile.java
@@ -2,6 +2,7 @@ package io.micronaut.gradle.docker;
 
 import com.bmuschko.gradle.docker.tasks.image.Dockerfile;
 import org.graalvm.buildtools.gradle.NativeImagePlugin;
+import org.graalvm.buildtools.gradle.dsl.AgentConfiguration;
 import org.graalvm.buildtools.gradle.dsl.NativeImageOptions;
 import org.graalvm.buildtools.gradle.dsl.NativeResourcesOptions;
 import org.graalvm.buildtools.gradle.internal.BaseNativeImageOptions;
@@ -202,8 +203,23 @@ public abstract class NativeImageDockerfile extends Dockerfile implements Docker
                             return delegate.getVerbose();
                         }
 
+                        /**
+                         * Configures the GraalVM agent options.
+                         *
+                         * @param spec the agent configuration.
+                         */
                         @Override
-                        public Property<Boolean> getAgent() {
+                        public void agent(Action<? super AgentConfiguration> spec) {
+                            delegate.agent(spec);
+                        }
+
+                        /**
+                         * Returns the GraalVM agent configuration.
+                         *
+                         * @return the configuration.
+                         */
+                        @Override
+                        public AgentConfiguration getAgent() {
                             return delegate.getAgent();
                         }
 

--- a/src/main/java/io/micronaut/gradle/graalvm/MicronautGraalPlugin.java
+++ b/src/main/java/io/micronaut/gradle/graalvm/MicronautGraalPlugin.java
@@ -53,7 +53,11 @@ public class MicronautGraalPlugin implements Plugin<Project> {
         });
         GraalVMExtension graal = project.getExtensions().findByType(GraalVMExtension.class);
         graal.getBinaries().configureEach(options ->
-                options.resources(rsrc -> rsrc.autodetection(inf -> inf.getEnabled().convention(true)))
+                options.resources(rsrc -> rsrc.autodetection(inf -> {
+                    inf.getEnabled().convention(true);
+                    inf.getIgnoreExistingResourcesConfigFile().convention(true);
+                    inf.getRestrictToProjectDependencies().convention(true);
+                }))
         );
         project.getPluginManager().withPlugin("application", plugin -> {
             TaskContainer tasks = project.getTasks();

--- a/src/main/java/io/micronaut/gradle/graalvm/MicronautGraalPlugin.java
+++ b/src/main/java/io/micronaut/gradle/graalvm/MicronautGraalPlugin.java
@@ -45,7 +45,11 @@ public class MicronautGraalPlugin implements Plugin<Project> {
         project.getPluginManager().apply(NativeImagePlugin.class);
         project.getPluginManager().withPlugin("io.micronaut.library", plugin -> {
             MicronautExtension extension = project.getExtensions().findByType(MicronautExtension.class);
-            configureMicronautLibrary(project, extension);
+            configureAnnotationProcessing(project, extension);
+        });
+        project.getPluginManager().withPlugin("io.micronaut.application", plugin -> {
+            MicronautExtension extension = project.getExtensions().findByType(MicronautExtension.class);
+            configureAnnotationProcessing(project, extension);
         });
         GraalVMExtension graal = project.getExtensions().findByType(GraalVMExtension.class);
         graal.getBinaries().configureEach(options ->
@@ -109,7 +113,7 @@ public class MicronautGraalPlugin implements Plugin<Project> {
         }
     }
 
-    private static void configureMicronautLibrary(Project project, MicronautExtension extension) {
+    private static void configureAnnotationProcessing(Project project, MicronautExtension extension) {
         SourceSetContainer sourceSets = project
                 .getConvention()
                 .getPlugin(JavaPluginConvention.class)

--- a/src/test/groovy/io/micronaut/gradle/AbstractGradleBuildSpec.groovy
+++ b/src/test/groovy/io/micronaut/gradle/AbstractGradleBuildSpec.groovy
@@ -65,6 +65,7 @@ abstract class AbstractGradleBuildSpec extends Specification {
                 .forwardStdOutput(System.out.newWriter())
                 .forwardStdError(System.err.newWriter())
                 .withPluginClasspath()
+                .withDebug(true)
                 .build()
     }
 

--- a/src/test/groovy/io/micronaut/gradle/MicronautGraalWellBehavePluginSpec.groovy
+++ b/src/test/groovy/io/micronaut/gradle/MicronautGraalWellBehavePluginSpec.groovy
@@ -1,0 +1,22 @@
+package io.micronaut.gradle
+
+import org.gradle.testfixtures.ProjectBuilder
+import spock.lang.Specification
+
+class MicronautGraalWellBehavePluginSpec extends Specification {
+    def "graal processor is added with plugin #plugin"() {
+        def project = ProjectBuilder.builder().build()
+
+        when:
+        project.plugins.apply (plugin)
+
+        then:
+        project.configurations.annotationProcessor.dependencies.find { it.name == 'micronaut-graal' }
+
+        where:
+        plugin << [
+                "io.micronaut.library",
+                "io.micronaut.application",
+        ]
+    }
+}

